src/wfc/collapse.rs

```diff
@@ -114,7 +117,8 @@ fn exec_collapse(
             to_propagate = replace(&mut propagations, to_propagate);
         }
 
-        if counter >= iterations { return vertices }
+        if progress { output_vertices.push(vertices.clone()) }
+        if counter >= iterations { return vec![vertices.clone()] }
```

Comment:
Why not just push `vertices` to `output_vertices` and return that? This way we loose all the progress vecs when we hit the number of iterations even if `progress = true`. Also, I don't think this `clone` should be needed.